shardingsphere-sql-parser/shardingsphere-sql-parser-postgresql/src/main/java/org/apache/shardingsphere/sql/parser/visitor/impl/PostgreSQLDALVisitor.java

```diff
@@ -33,26 +42,50 @@
  * DAL visitor for PostgreSQL.
  */
 public final class PostgreSQLDALVisitor extends PostgreSQLVisitor implements DALVisitor {
-    
+
     @Override
     public ASTNode visitShow(final ShowContext ctx) {
         return new ShowStatement();
     }
-    
+
     @Override
-    public ASTNode visitSet(final SetContext ctx) {
+    public ASTNode visitSetVariable(final SetVariableContext ctx) {
         SetStatement result = new SetStatement();
-        if (null != ctx.configurationParameterClause()) {
-            result.setVariable((VariableSegment) visit(ctx.configurationParameterClause()));
+        if (null != ctx.variableExpr()) {
+            List<VariableExpr> variableExprList = new ArrayList<VariableExpr>(1);
```

Comment:
If the size of ArrayList is 1, why we need create a list?
How about return the element directly?

Reply:
For  future support pgsql some variable's condition and pgsql and mysql unity in code

Reply:
Use `Collections.singletionList(xxx)` is better

Reply:
done